__tests__/unit/react/Modal.test.js

```diff
@@ -1,3 +1,34 @@
-it("test sample", () => {
-  expect(1).toBe(1);
+import React from "react";
+import Modal from "../../../src/components/Modal";
+
+import { render, screen } from "@testing-library/react";
+import "@testing-library/jest-dom";
+
+describe("Modal", () => {
+  it("nodeId가 제공되면 children을 렌더링합니다.", () => {
+    const testText = "Tree Modal";
+    render(<Modal nodeId={1}>{testText}</Modal>);
+
+    const contentElement = screen.getByText(testText);
+    expect(contentElement).toBeInTheDocument();
+
+    const computedStyle = window.getComputedStyle(contentElement.parentElement);
+    expect(computedStyle.display).toBe("block");
+  });
+
+  it("nodeId가 제공되지 않으면 children을 렌더링하지 않습니다.", () => {
+    const testText = "Tree Modal";
+    render(<Modal nodeId={null}>{testText}</Modal>);
+
+    const contentElement = screen.queryByText(testText);
+
+    if (contentElement) {
+      const computedStyle = window.getComputedStyle(
+        contentElement.parentElement,
+      );
+      expect(computedStyle.display).toBe("block");
+    } else {
+      expect(contentElement).not.toBeInTheDocument();
+    }
+  });
```

Comment:
children을 렌더링하지 않는 것을 확인해야하기 때문에 `expect(computedStyle.display).toBe("block");` 이 구문은 안 써도 되지 않을까요?

```suggestion
  it("nodeId가 제공되지 않으면 children을 렌더링하지 않습니다.", () => {
    const testText = "Tree Modal";
    render(<Modal nodeId={null}>{testText}</Modal>);

    const contentElement = screen.queryByText(testText);

    setTimeout(() => {
      expect(contentElement).not.toBeInTheDocument();
    }, 0);
  });
```
렌더링 이후에 `expect(contentElement).not.toBeInTheDocument();`를 실행하기 위에 `setTimeout()`을 사용했는데, 비동기적인 흐름을 테스트하기 위한 최선의 방법은 아닐 수도 있습니다.. 이 부분은 저도 좀 더 조사해보겠습니다.